modules/fgbio/fastqtobam/main.nf

```diff
@@ -9,29 +9,31 @@ process FGBIO_FASTQTOBAM {
 
     input:
     tuple val(meta), path(reads)
-    val read_structure
 
     output:
-    tuple val(meta), path("*_umi_converted.bam"), emit: umibam
-    path "versions.yml"                         , emit: versions
+    tuple val(meta), path("*.bam") , emit: bam , optional: true
+    tuple val(meta), path("*.cram"), emit: cram, optional: true
+    path "versions.yml"            , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def sample_name = args.contains("--sample") ? "" : "--sample ${prefix}"
+    def library_name = args.contains("--library") ? "" : "--library ${prefix}"
```

Comment:
I see an issue with these, when the prefix is already named `.bam` or `.cram` then this will also be the sample and library name. I think that should either be stripped or use an optional input to toggle CRAM. What do you think?

Reply:
I noticed that too, but when you're already adding option to modify the `$prefix` you might as well set `--sample` and `--library` in `ext.args` IMO.

Reply:
I added a test where both are set explicitly to test for this case

Reply:
Perfect, I only commented on this because you were _not_ configuring it that way in the tests 😉

Reply:
yeah, I did use separate tests. I probably should've done it all in one.